sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/MesosResourcePool.java

```diff
@@ -18,15 +18,19 @@
 public class MesosResourcePool {
     private static final Logger logger = LoggerFactory.getLogger(MesosResourcePool.class);
 
-    private final Offer offer;
-    private final Map<String, List<MesosResource>> unreservedAtomicPool;
-    private final Map<String, Value> unreservedMergedPool;
-    private final Map<String, MesosResource> reservedPool;
+    private Offer offer;
+    private Map<String, List<MesosResource>> unreservedAtomicPool;
+    private Map<String, Value> unreservedMergedPool;
+    private Map<String, MesosResource> reservedPool;
 
     /**
      * Creates a new pool of resources based on what's available in the provided {@link Offer}.
      */
     public MesosResourcePool(Offer offer) {
+        init(offer);
+    }
+
+    private void init(Offer offer) {
```

Comment:
Not easily -- the pipeline keeps the same `MesosResourcePool` and passes it to each stage in turn, so this change would get lost to later stages, since the evaluation stage in question would just overwrite its local reference to that pool with a new reference, rather than the original object being mutated.